build.gradle.kts

```diff
@@ -46,10 +46,31 @@ allprojects {
         }
     }
 
+    val setAllWarningsAsError = providers.gradleProperty("kotlin_Werror_override").map {
+        when (it) {
+            "enable" -> true
+            "disable" -> false
+            else -> error("Unexpected value for 'kotlin_Werror_override' property: $it")
+        }
+    }
+
     tasks.withType(KotlinCompilationTask::class).configureEach {
         compilerOptions {
-            allWarningsAsErrors = true
-            freeCompilerArgs.add("-Xexpect-actual-classes")
+            if (setAllWarningsAsError.orNull != false) {
+                allWarningsAsErrors = true
+            } else {
+                freeCompilerArgs.addAll(
+                    "-Wextra",
+                    "-Xuse-fir-experimental-checkers"
+                )
+                logger.info("kotlin_Werror_override was set to 'disable'")
```

Comment:
@woainikk that'll emit a message for each and every compilation task, but I think it's better than writing it outside tasks configuration and them accidentally forgetting to setup `freeCompilerArgs`.